internal/store/access_role.go

```diff
@@ -19,16 +19,17 @@ import (
 
 	"github.com/mongodb/mongodb-atlas-cli/internal/config"
 	atlas "go.mongodb.org/atlas/mongodbatlas"
+	atlasv2 "go.mongodb.org/atlas/mongodbatlasv2"
 )
 
 //go:generate mockgen -destination=../mocks/mock_access_role.go -package=mocks github.com/mongodb/mongodb-atlas-cli/internal/store CloudProviderAccessRoleCreator,CloudProviderAccessRoleAuthorizer,CloudProviderAccessRoleLister,CloudProviderAccessRoleDeauthorizer
 
 type CloudProviderAccessRoleCreator interface {
-	CreateCloudProviderAccessRole(string, string) (*atlas.AWSIAMRole, error)
+	CreateCloudProviderAccessRole(string, string) (*atlasv2.CloudProviderAccessRole, error)
```

Comment:
nit - we might decide to use v2 as shortcut to make functions shorter. Just though